### PR TITLE
5124: Manage Additional Services

### PR DESCRIPTION
**Pull request by PhysicsJon**
[AB#5124](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/5124)
[AB#8154](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/8154)

### CODE REVIEW

File: src/OrderFormAcceptanceTests.TestData/Order.cs [hunk 1]
```diff
@@ -140,6 +140,18 @@ public void Delete(string connectionString)
...
+            var query = @"DELETE FROM [dbo].[OrderItem] WHERE OrderId=@orderId";
+            SqlExecutor.Execute<Order>(connectionString, query, this);
+		}
+
+        public void DeleteAllServiceRecipientsForOrderId(string connectionString)
+        {
+            var query = @"DELETE FROM [dbo].[ServiceRecipient] WHERE OrderId=@orderId";
+            SqlExecutor.Execute<Order>(connectionString, query, this);
+        }
+
```
Comment (talawson05) [inline src/OrderFormAcceptanceTests.TestData/Order.cs:154]: I don't think these two methods should be in the Order class, and why do you need them? The hooks currently do this anyway as part of tear down

File: src/OrderFormAcceptanceTests.Tests/Features/ManageOrderForm/AdditionalServices.feature [hunk 2]
```diff
@@ -4,36 +4,63 @@
...
+Scenario: No Additional Service added
+	When the Order Form for the existing order is presented
+	And the User has chosen to manage the Additional Service section
+	And there is no Additional Service added to the order
+	Then there is content indicating there is no Additional Service added
+
+Scenario: User chooses not to add Additional Service
+	When the Order Form for the existing order is presented
+	And the User has chosen to manage the Additional Service section
+	And there is no Additional Service added to the order
```
Comment (talawson05) [inline src/OrderFormAcceptanceTests.Tests/Features/ManageOrderForm/AdditionalServices.feature:58]: The three steps here should be Givens/Ands - make the underlying step a generic StepDefinition type so you can use Given/When in the feature files

File: src/OrderFormAcceptanceTests.Tests/Features/ManageOrderForm/AdditionalServices.feature [hunk 3]
```diff
@@ -4,36 +4,63 @@
...
+	And the User has chosen to manage the Additional Service section
+	Then the Additional Service dashboard is presented
+	And the Order description is displayed
+	And there is a control to add a Additional Service
+	And there is a control to continue
+
+Scenario: No Additional Service added
+	When the Order Form for the existing order is presented
+	And the User has chosen to manage the Additional Service section
+	And there is no Additional Service added to the order
```
Comment (talawson05) [inline src/OrderFormAcceptanceTests.Tests/Features/ManageOrderForm/AdditionalServices.feature:52]: this is your When step, the other two are Givens

File: src/OrderFormAcceptanceTests.Tests/Features/ManageOrderForm/AdditionalServices.feature [hunk 4]
```diff
@@ -4,36 +4,63 @@
...
+	When they choose to continue	
+	Then the content validation status of the additional-services section is complete
+
+Scenario: Go back
+	When the Order Form for the existing order is presented
+	And the User has chosen to manage the Additional Service section
+	And the User chooses to go back
```
Comment (talawson05) [inline src/OrderFormAcceptanceTests.Tests/Features/ManageOrderForm/AdditionalServices.feature:65]: This is the When step, the other two are Givens

File: src/OrderFormAcceptanceTests.Tests/Features/ManageOrderForm/AdditionalServices.feature [hunk 5]
```diff
@@ -4,36 +4,63 @@
...
+	Then there is content indicating there is no Additional Service added
+
+Scenario: User chooses not to add Additional Service
+	When the Order Form for the existing order is presented
+	And the User has chosen to manage the Additional Service section
+	And there is no Additional Service added to the order
+	When they choose to continue	
+	Then the content validation status of the additional-services section is complete
+
+Scenario: Go back
```
Comment (talawson05) [inline src/OrderFormAcceptanceTests.Tests/Features/ManageOrderForm/AdditionalServices.feature:62]: Can you prefix all scenario titles with the feature file name

File: src/OrderFormAcceptanceTests.TestData/Order.cs (hunk 1 above)
Comment (PhysicsJon) [reply src/OrderFormAcceptanceTests.TestData/Order.cs:154]: I think it cleans up the hooks method, prevents the need for loops to delete all SR's and Catalogue Items. I could move them elsewhere, but as they run from the OrderId, I felt it was better to have them within the Order class

Comment (talawson05) [reply src/OrderFormAcceptanceTests.TestData/Order.cs:154]: It does look cleaner, but let's put them in the OrderItem and ServiceRecipient objects and pass the orderId as a parameter

**Pull request merged**